matching-notification-rest/src/main/java/org/phenotips/matchingnotification/rest/MatchingNotificationResource.java

```diff
@@ -0,0 +1,153 @@
+     * @param genScore only matches with genotypic score higher or equal to this value are returned
+     * @param onlyNotified when true only loads matches that have been notified
+     * @return a response containing a JSON object with a list of matches
+     */
+    @POST
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    @Path("/show-matches")
+    Response getMatches(@FormParam("score") @DefaultValue("0.5") double score,
+        @FormParam("phenScore") @DefaultValue("0") double phenScore,
+        @FormParam("genScore") @DefaultValue("0.1") double genScore,
+        @FormParam("onlyNotified") @DefaultValue("false") boolean onlyNotified);
+
+    /**
+     * Sends email notifications for each match. Example:
+     *
+     * <pre>
+     *  Input JSON: {"ids": [{'matchId' : 58, 'patientId' : P000067}, {'matchId' : 7, 'patientId' : P000035}]}
+     *  Output JSON: {"results": {"success":[a list of match ids], "failed": [a list of match ids]}}
+     * </pre>
+     * <dl>
+     * <dt>ids</dt>
+     * <dd>a JSON object (in the format as described above) with a list of match ids that given patients ids
+     * should be notified about</dd>
+     * </dl>
+     *
+     * @return result JSON
+     */
+    @POST
+    @Produces(MediaType.APPLICATION_JSON)
+    @Path("/send-notifications")
+    Response sendNotifications();
+
+    /**
+     * Returns the content of email to be sent when a user hits "contact this match" button.
+     *
+     * @param matchId the id of the match that should be used to generate the email
+     * @param patientId the id of the patient that is the subject of the email (can be either of
+     *                  the two patients involved in the specified match)
+     * @return a response containing a JSON object, in the following format:
+     *     <pre>
+     *      { "emailContent": text,
+     *        "recipients": [listof email addresses as strings],
+     *        "contentType": type,
+     *        "subject": text }
+     *     </pre>
+     *     where text is a string, and type is the type of content as string (e.g. "text/plain")
+     *     TODO: check what other types we may be returning
+     */
+    @POST
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    @Path("/preview-match-email")
+    Response getEmailToBeSent(@FormParam("matchId") String matchId,
+            @FormParam("subjectPatientId") String patientId);
+
+    /**
+     * Marks matches, with ids given in parameter, as saved, rejected or uncategorized. Example:
+     *
+     * <pre>
+     * Input: ["1", "2"], "saved"
+     * Output: {"results": {"success": [1,2]}}
+     * </pre>
+     *
+     * @param matchesIds List of matches IDs to change the status for
+     * @param status whether matches status should be set as 'saved', 'rejected' or 'uncategorized'
+     * @return result JSON
+     */
+    @POST
+    @Consumes(MediaType.APPLICATION_JSON)
```

Comment:
Does it consume anything? If form params are expected, then the input should be `form-encoded`, not `JSON`.